authorization/src/main/java/com/terabite/programming/controller/ProgrammingControler.java

```diff
@@ -40,43 +51,86 @@ public ProgrammingControler(RepCycleService repCycleService, DayService dayServi
     
     //Program endpoints
     @PostMapping("/program")
-    public ResponseEntity<?> postProgram(@RequestBody Program program) {
+    public ResponseEntity<?> postProgram(@RequestBody CreateProgramRequest request) {
+        Program program = new Program(request.programName(), List.of());
         return programService.createNewProgram(program);
     }
 
     @PutMapping("/program")
-    public ResponseEntity<?> putProgram(@RequestBody Program program) {
-        return programService.updateProgram(program);
+    public ResponseEntity<?> updateProgram(@RequestBody UpdateProgramRequest program) {
+        List<Week> weeks = List.of();
```

Comment:
`List.of` creates a read-only list - attempting to adding elements will throw an exception